jib-cli/src/test/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessorTest.java

```diff
@@ -417,4 +423,318 @@ public void testComputeEntrypointForPackagedStandard_withMainClass()
+    assertThat(loaderLayer.getName()).isEqualTo("spring-boot-loader");
+    assertThat(
+            loaderLayer
+                .getEntries()
+                .stream()
+                .map(FileEntry::getExtractionPath)
+                .collect(Collectors.toList()))
+        .containsExactly(
+            AbsoluteUnixPath.get("/app/BOOT-INF"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes/classDirectory"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/lib"),
+            AbsoluteUnixPath.get("/app/META-INF"),
+            AbsoluteUnixPath.get("/app/org"),
+            AbsoluteUnixPath.get("/app/org/springframework"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader/data"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader/data/data1.class"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader/launcher1.class"));
+
+    assertThat(snapshotLayer.getName()).isEqualTo("snapshot-dependencies");
+    assertThat(
+            snapshotLayer
+                .getEntries()
+                .stream()
+                .map(FileEntry::getExtractionPath)
+                .collect(Collectors.toList()))
+        .containsExactly(
+            AbsoluteUnixPath.get("/app/BOOT-INF"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes/classDirectory"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/lib"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/lib/dependency3-SNAPSHOT.jar"),
+            AbsoluteUnixPath.get("/app/META-INF"),
+            AbsoluteUnixPath.get("/app/org"),
+            AbsoluteUnixPath.get("/app/org/springframework"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader"),
+            AbsoluteUnixPath.get("/app/org/springframework/boot/loader/data"));
+
+    assertThat(applicationLayer.getName()).isEqualTo("application");
+    assertThat(
+            applicationLayer
+                .getEntries()
+                .stream()
+                .map(FileEntry::getExtractionPath)
+                .collect(Collectors.toList()))
+        .containsExactly(
+            AbsoluteUnixPath.get("/app/BOOT-INF"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes/class1.class"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes/classDirectory/class2.class"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/classes/classDirectory"),
+            AbsoluteUnixPath.get("/app/BOOT-INF/lib"),
+            AbsoluteUnixPath.get("/app/META-INF"),
+            AbsoluteUnixPath.get("/app/META-INF/MANIFEST.MF"),
```

Comment:
Interesting. I guess `layer.idx` in this test doesn't have a resource layer for `META-INF`?

Reply:
The layer.idx has an "application" layer which contains both classes and resources.